### PR TITLE
DAOS-7851 vos: return value checking

### PR DESCRIPTION
**Pull request by NiuYawei**
Fix few missing return value checking cases.

Signed-off-by: Niu Yawei <yawei.niu@intel.com>

### CODE REVIEW

File: src/common/btree.c [hunk 1]
```diff
@@ -845,8 +851,14 @@ btr_root_start(struct btr_context *tcx, struct btr_record *rec)
...
 
-	if (btr_has_tx(tcx))
-		btr_root_tx_add(tcx); /* XXX check error */
+	if (btr_has_tx(tcx)) {
+		rc = btr_root_tx_add(tcx);
+		if (rc != 0) {
+			D_ERROR("Failed to add root into TX: %s\n",
+				strerror(errno));
+			return rc;
+		}
```
Comment (liw) [inline src/common/btree.c:860]: In the volatile memory (i.e., DRAM) case, it looks like we would need to roll back the allocation done above by btr_node_alloc.

File: src/common/btree.c [hunk 2]
```diff
@@ -901,8 +913,14 @@ btr_root_grow(struct btr_context *tcx, umem_off_t off_left,
...
 	/* replace the root offset, increase tree level */
-	if (btr_has_tx(tcx))
-		btr_root_tx_add(tcx); /* XXX check error */
+	if (btr_has_tx(tcx)) {
+		rc = btr_root_tx_add(tcx); /* XXX check error */
+		if (rc != 0) {
+			D_ERROR("Failed to add root into TX: %s\n",
+				strerror(errno));
+			return rc;
+		}
```
Comment (liw) [inline src/common/btree.c:922]: Same as my comment on btr_root_start.

**Review by liw — APPROVED** [review]

File: src/common/btree.c (hunk 1 above)
Comment (NiuYawei) [reply src/common/btree.c:860]: Indeed, I'm not sure if there are other places missing rollback for vmem tree, and not sure if it's easy to rollback for vmem tree for every failure cases.  @jolivier23 any idea? should we open a new ticket to scrutinize missing rollback for vmem tree case?

Comment (jolivier23) [reply src/common/btree.c:860]: umem_tx_add_ptr returns 0 for vmem so this should never happen.

File: src/common/btree.c (hunk 2 above)
Comment (jolivier23) [reply src/common/btree.c:922]: Same comment.  Perhaps we could assert rc == 0 || btr_umm(tcx)->umm_id != UMEM_CLASS_VMEM

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**